src/bulletproof/party.rs

```diff
@@ -0,0 +1,323 @@
+    /// Assigns a position in the aggregated proof to this party,
+    /// allowing the party to commit to the bits of their value.
+    #[cfg(test)]
+    pub fn assign_position(
+        self,
+        j: usize,
+    ) -> Result<(PartyAwaitingBitChallenge<'a>, BitCommitment), MPCError> {
+        self.assign_position_with_rng(j, &mut rand::thread_rng())
+    }
+
+    /// Assigns a position in the aggregated proof to this party,
+    /// allowing the party to commit to the bits of their value.
+    pub fn assign_position_with_rng<T: RngCore + CryptoRng>(
+        self,
+        j: usize,
+        rng: &mut T,
+    ) -> Result<(PartyAwaitingBitChallenge<'a>, BitCommitment), MPCError> {
+        if self.bp_gens.party_capacity <= j {
+            return Err(MPCError::InvalidGeneratorsLength);
+        }
+
+        let bp_share = self.bp_gens.share(j);
+
+        let a_blinding = Scalar::random(rng) * *INV_EIGHT;
+
+        // Compute A = <a_L, G> + <a_R, H> + a_blinding * B_blinding
+        let mut A = self.pc_gens.B_blinding * a_blinding * *INV_EIGHT;
+
+        use subtle::{Choice, ConditionallySelectable};
+        let mut i = 0;
+        for (G_i, H_i) in bp_share.G(self.n).zip(bp_share.H(self.n)) {
+            // If v_i = 0, we add a_L[i] * G[i] + a_R[i] * H[i] = - H[i]
+            // If v_i = 1, we add a_L[i] * G[i] + a_R[i] * H[i] =   G[i]
+            let v_i = Choice::from(((self.v >> i) & 1) as u8);
+            let mut point = -H_i;
+            point.conditional_assign(G_i, v_i);
+            A += point * *INV_EIGHT;
+            i += 1;
+        }
+
+        let s_blinding = Scalar::random(rng);
+        let s_L: Vec<Scalar> = (0..self.n).map(|_| Scalar::random(rng)).collect();
+        let s_R: Vec<Scalar> = (0..self.n).map(|_| Scalar::random(rng)).collect();
+
+        // NOTE: We did not verify this, but it looks exactly the same
+        // Compute S = <s_L, G> + <s_R, H> + s_blinding * B_blinding
+        let S = EdwardsPoint::multiscalar_mul(
+            iter::once(&s_blinding).chain(s_L.iter()).chain(s_R.iter()),
+            iter::once(&self.pc_gens.B_blinding)
+                .chain(bp_share.G(self.n))
+                .chain(bp_share.H(self.n)),
+        );
+        let S = S * *INV_EIGHT;
+
+        // Return next state and all commitments
+        let bit_commitment = BitCommitment {
+            V_j: self.V,
+            A_j: A,
+            S_j: S,
+        };
+        let next_state = PartyAwaitingBitChallenge {
+            n: self.n,
+            v: self.v,
+            v_blinding: self.v_blinding,
+            pc_gens: self.pc_gens,
+            j,
+            a_blinding,
+            s_blinding,
+            s_L,
+            s_R,
+        };
+        Ok((next_state, bit_commitment))
+    }
+}
+
+/// Overwrite secrets with null bytes when they go out of scope.
+impl<'a> Drop for PartyAwaitingPosition<'a> {
+    fn drop(&mut self) {
+        self.v.clear();
+        self.v_blinding.clear();
+    }
+}
```

Comment:
I know this probably copied over and I am not expert on this but I think there is likely a reason why the `zeroize` crate exists because I assume it is not that easy :D